test/core/end2end/fake_resolver.h

```diff
@@ -32,8 +32,23 @@
 #ifndef GRPC_TEST_CORE_END2END_FAKE_RESOLVER_H
 #define GRPC_TEST_CORE_END2END_FAKE_RESOLVER_H
 
+#include "src/core/lib/channel/channel_args.h"
 #include "test/core/util/test_config.h"
 
 void grpc_fake_resolver_init();
 
+// Return a \a grpc_arg enabling LB.
+grpc_arg grpc_fake_resolver_lb_enabled_arg();
```

Comment:
The approach you've implemented here is certainly fine for now.  However, I think that ultimately, it would be better to have a more flexible interface.  For example, we can define a new object called something like `grpc_fake_resolver_response_generator`.  The object will have a method that will be used to pass in a set of channel args (including the resolved address arg) to be returned by the fake resolver.  That approach will allow us to test things like (a) the resolver returning some backend address and some balancer addresses and (b) the resolver returning a new list of addresses after the LB channel is already warmed up.